### PR TITLE
Installation assistant test and tier workflow migration 

### PR DESCRIPTION
**Pull request by davidcr01**
## Description

Related: #20 
The aim of this PR is to migrate the `Test_unattended` Jenkins pipeline to GHA. This PR adds a new workflow, `Test_installation_assistant.yml` which performs an AIO installation in the selected system.

Because of the matrix strategy, the `Test_unattended_tier` pipeline has been included in the new workflow, launching a job per selected system.

In addition to migrating the pipeline itself, different features have also been added:
- If the base instance has been created, it is destroyed if the workflow fails or is canceled, as long as the destroy option is checked.
- If the destroy option is not checked, an artifact is generated with the necessary data to access the instance, such as the inventory or the private key that is generated with the execution of the allocator.

## Testing (single system)

The performed testing is the following:
- :heavy_check_mark: CentOS 7: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683486737
- :heavy_check_mark: CentOS 8: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683489080
- :heavy_check_mark: RHEL7: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683458991
- :heavy_check_mark: RHEL8: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683463848
- :heavy_check_mark: Ubuntu 20: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683452138
- :heavy_check_mark: Ubuntu 22: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683456409
- :heavy_check_mark: Ubuntu 18: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683448786
- :heavy_check_mark: Ubuntu 16: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683445779

> [!WARNING]
> The AL2 system does not finish in success because of: https://github.com/wazuh/wazuh/issues/25534, but it completes almost every step of the workflow: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10683582122

## Testing - using the tier

- In the following workflow run, each job represents a test in a specific and selected OS: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10715530117
- In this case, when unselecting the `DESTROY` option, the artifacts uploaded are unique: https://github.com/wazuh/wazuh-installation-assistant/actions/runs/10715858016

### CODE REVIEW

**Review by c-bordon — APPROVED** [review]

**Review by CarlosALgit — DISMISSED** [review]

File: .github/workflows/ansible-playbooks/provision.yml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    - name: Clone installation assistant git repository
+      git:
+        repo: "{{ repository }}"
+        dest: "{{ tmp_path }}"
+        version: "{{ reference }}"
+        depth: 1
+        force: true
+
+    - name: Generate unattended
```
Comment (teddytpc1) [inline .github/workflows/ansible-playbooks/provision.yml:172]: Change the `unattended` references for `Installation assistant`.

File: .github/workflows/ansible-playbooks/aio_tests.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+  - hosts: all
+    become: true
+
+    vars:
+      script_path: "{{ tmp_path }}/tests/install"
+      script_name: "{{ test_name }}.py"
+    
+    tasks:
+      - name: Test unattended AIO install
```
Comment (teddytpc1) [inline .github/workflows/ansible-playbooks/aio_tests.yml:10]: Change the `unattended` references for `Installation assistant`.

**Review by teddytpc1 — CHANGES_REQUESTED** [review]

File: .github/workflows/ansible-playbooks/provision.yml (hunk 1 above)
Comment (davidcr01) [reply .github/workflows/ansible-playbooks/provision.yml]: Done

File: .github/workflows/ansible-playbooks/aio_tests.yml (hunk 2 above)
Comment (davidcr01) [reply .github/workflows/ansible-playbooks/aio_tests.yml]: Done

**Review by teddytpc1 — APPROVED** [review]

**Review by c-bordon — APPROVED** [review]

**Pull request merged**